mlflow/sklearn/utils.py

```diff
@@ -126,16 +133,38 @@ def _get_metrics_value_dict(metrics_list):
         try:
             metric_value = metric.function(**metric.arguments)
         except Exception as e:  # pylint: disable=broad-except
-            _log_warning_for_metrics(metric.name, metric.function, e)
+            _log_warning_for_metrics_or_artifacts(metric.name, metric.function, e, True)
         else:
             metric_value_dict[metric.name] = metric_value
-
     return metric_value_dict
 
 
+# Remove duplicates from y_true while still preserving orders
+def _get_labels(y_true):
+    seen = set()
+    return [x for x in y_true if not (x in seen or seen.add(x))]
```

Comment:
why do we need to preserve labels order?

Reply:
My thought:
As we only take in X and Y from users, sometimes I need to check whether this is a binary classifier problem so that to log respective metrics. 
Get the non-duplicated labels can help me to decide whether this is a binary problem or not.

Reply:
Right, but I don't think we need to preserve any sort of ordering to perform this checking (which is @harupy's point). Because this routine is iterating over a list and adding elements to a set, I don't think it's order-preserving anyway. 

It should be sufficient to simplify this by returning `set(y_true)`. At that point, `_get_labels` is redundant because it simply wraps `set`. Accordingly, I think we can just remove this function and use `set()` where this used to be called.